src/renderer/api/kube-object.ts

```diff
@@ -267,6 +280,12 @@ export class KubeObject<Metadata extends IKubeObjectMetadata = IKubeObjectMetada
 
   // use unified resource-applier api for updating all k8s objects
   async update<T extends KubeObject>(data: Partial<T>): Promise<T> {
+    for (const field of KubeObject.nonEditableFields) {
+      if (!_.isEqual(_.get(this, field), _.get(data, field))) {
```

Comment:
```suggestion
      if (this[field] !== data[field]) {
```

Reply:
There is no need to use lodash here

Reply:
I disagree, if a user edits a deep field back to the same value we don't want to report an error.

Reply:
Sorry, I didn't look closely enough. You are right.